김현수/mvvmstudy/app/src/main/java/com/example/mvvmstudy/ViewModel/DetailViewModel.kt

```diff
@@ -0,0 +1,21 @@
+package com.example.mvvmstudy.ViewModel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import com.example.mvvmstudy.data.Detail
+
+
+class DetailViewModel(savedStateHandle: SavedStateHandle) : ViewModel() {
+
+    private val _detail: Detail? = savedStateHandle.get<Detail>("detail")
+
+    private val _ui: MutableLiveData<Detail> = MutableLiveData()
+    val ui: LiveData<Detail>
+        get() = _ui
+
+    init {
+        _ui.value = _detail!!
+    }
```

Comment:
위 과정을 한 줄로 요약할 수 있을 것 같아요!
```suggestion
   val detail = savedStateHandle.getLiveData<Detail>("detail")
```

Reply:
그렇군요 ㅎㅎ 감사합니다.